### PR TITLE
Aruco Support for AprilTag Detection (Experimental)

### PR DESCRIPTION
**Pull request by mdurrani808**
Uses OpenCV's aruco module for AprilTag detection.

### CODE REVIEW

File: photon-server/src/main/resources/web/index.html [hunk 1]
```diff
@@ -1 +1 @@
-<p>UI has not been copied!</p>
+<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width,initial-scale=1"><link rel="icon" href="/favicon.png"><title>PhotonVision</title><link href="/js/chunk-2d216214.90e8fa62.js" rel="prefetch"><link href="/js/chunk-2d216257.8e8f2e7b.js" rel="prefetch"><link href="/js/chunk-fdf80f12.244b6d05.js" rel="prefetch"><link href="/css/app.53f364c5.css" rel="preload" as="style"><link href="/css/chunk-vendors.a79457dc.css" rel="preload" as="style"><link href="/js/app.8e11f439.js" rel="preload" as="script"><link href="/js/chunk-vendors.7c9bd682.js" rel="preload" as="script"><link href="/css/chunk-vendors.a79457dc.css" rel="stylesheet"><link href="/css/app.53f364c5.css" rel="stylesheet"></head><body><noscript><strong>We're sorry but PhotonVision doesn't work properly without JavaScript enabled. Please enable it to continue.</strong></noscript><div id="app"></div><script src="/js/chunk-vendors.7c9bd682.js"></script><script src="/js/app.8e11f439.js"></script></body></html>
```
Comment (mcm001) [inline photon-server/src/main/resources/web/index.html:1]: ope

File: photon-thinclient/thinclient.html [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+>>>>>>> 9d03566c09647195a559414d383e85766e0574b0
+            }
+
+            ws_connect() {
+                this.ws = new WebSocket(this.serverAddr);
+<<<<<<< HEAD
+                this.ws.binaryType = "arraybuffer";
+=======
+                this.ws.binaryType = "blob";
+>>>>>>> 9d03566c09647195a559414d383e85766e0574b0
```
Comment (mcm001) [inline photon-thinclient/thinclient.html:242]: lmao

File: photon-server/src/main/resources/web/index.html (hunk 1 above)
Comment (mdurrani808) [reply photon-server/src/main/resources/web/index.html:1]: ...

File: photon-thinclient/thinclient.html [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+
+<head>
+    <title>ThinClient</title>
+
+    <style>
+        * {
+            margin: 0;
+            padding: 0;
+        }
+        .imgbox {
```
Comment (mcm001) [inline photon-thinclient/thinclient.html:12]: I think the thinclient folder needs to be removed...>?

File: photon-core/src/main/java/org/photonvision/vision/target/TrackedTarget.java [hunk 4]
```diff
@@ -128,6 +130,60 @@ public TrackedTarget(DetectionResult result, TargetCalculationParameters params)
...
+                    -bestPose.getTranslation().getZ(),
+                    bestPose.getTranslation().getX()
+                });
+        setCameraRelativeTvec(tvec);
+
+        // Opencv expects a 3d vector with norm = angle and direction = axis
+        Rotation3d newRot = new Rotation3d(bestPose.getRotation().getY(),bestPose.getRotation().getZ(), -bestPose.getRotation().getX());
+        var rvec = new Mat(3, 1, CvType.CV_64FC1);
+        MathUtils.rotationToOpencvRvec(newRot, rvec);
+        setCameraRelativeRvec(rvec);
```
Comment (mcm001) [inline photon-core/src/main/java/org/photonvision/vision/target/TrackedTarget.java:184]: This smells a bit like a coordinate transform to me. Could we use the proper CoordinateSystem.convert call here, instead?

File: photon-core/src/main/java/org/photonvision/vision/aruco/PhotonArucoDetector.java [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+    public ArucoDetectionResult[] detect(
+            Mat grayscaleImg, CameraCalibrationCoefficients coeffs, ArucoDetector detector) {
+
+        detector.detectMarkers(grayscaleImg, corners, ids);
+
+        if (coeffs != null)
+            Aruco.estimatePoseSingleMarkers(
+                    corners,
+                    (float) Units.inchesToMeters(6),
```
Comment (mcm001) [inline photon-core/src/main/java/org/photonvision/vision/aruco/PhotonArucoDetector.java:75]: Not sure where this lives but having it hard coded here seems not ideal

File: photon-server/src/main/resources/web/index.html (hunk 1 above)
Comment (mdurrani808) [reply photon-server/src/main/resources/web/index.html]: How do I stop this from updating every time? Just adding it to gitignore?

File: photon-thinclient/thinclient.html (hunk 3 above)
Comment (mdurrani808) [reply photon-thinclient/thinclient.html:12]: Just adding it to gitignore or something else?

File: photon-core/src/main/java/org/photonvision/vision/target/TrackedTarget.java (hunk 4 above)
Comment (mdurrani808) [reply photon-core/src/main/java/org/photonvision/vision/target/TrackedTarget.java:184]: Yeah

**Review by mcm001 — APPROVED** [review]

**Review by mcm001 — APPROVED** [review]

**Pull request merged**